contents/why_julia.md

```diff
@@ -69,32 +74,271 @@ Let's proceed then!
+@fig:benchmarks was taken from [Julia's website benchmarks section^[please note that the Julia results depicted above do not include compile time.]](https://julialang.org/benchmarks/).
+As you can see Julia is **indeed** fast.
+
+![Julia versus other programming languages](images/benchmarks.png){#fig:benchmarks}
+
+We really believe in Julia (otherwise we wouldn't be writing this book).
+We think that Julia is the **future in scientific computing and scientific data analysis**.
+It enables the user to develop rapid and powerful code with a simple syntax.
+Usually, researchers develop code by prototyping using a very easy, but slow, language.
+Once the code is assured to run correct and fulfill its goal, then begins the process of converting the code to a fast, but hard, language.
+This is known as the "Two-Language Problem" and we discuss next.
 
 ### The Two-Language Problem {#sec:two_language}
 
-```{=comment}
-Alan Edelman TEDX talk @tedxtalksProgrammingLanguageHeal2020
-```
+The "Two-Language Problem" is a very typical situation in scientific computing where a researcher devises an algorithm or a solution to tackle a desired problem or analysis at hand.
+The solution is, then, prototyped in an easy to code language (like Python or R).
+If the prototype works, the researcher  would code in a fast language that would not be easy to prototype (C++ or FORTRAN).
+Thus, we have two languages involved in the process of of developing a new solution.
+One which is easy to prototype but is not suited for implementation (mostly due to being slow).
+And another one which is not so easy to code (and, consequently, not easy to prototype) but suited for implementation (mostly because it is fast).
+Julia comes to eliminate such situations by being the **same language that you prototype (ease of use) and implement the solution (speed)**.
+
+Also, Julia lets you use **unicode characters as variables or parameters**.
+This means no more using `sigma` or `sigma_i`, and instead just use `σ` or `σᵢ` as you would in mathematical notation.
+When you see code for an algorithm or for a mathematical equation you see almost the same notation and idiom.
+We call this feature **"One-To-One Code and Math Relation"**, while usually downplayed can be a powerful feature.
+
+We think that the "Two-Language problem" and the o"One-To-One Code and Math Relation" are best described by one of the creators of Julia, Alan Edelman, in a TEDx Talk [@tedxtalksProgrammingLanguageHeal2020] (if you reading the printed book or a static PDF please click on the link to go the video or check the citation):
+
+<iframe width="698" height="393" src="https://www.youtube.com/embed/qGW0GT1rCvs" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
```

Comment:
This is AMAZING! I really like how the book comes alive when we embed these short clips.

If I'm not mistaken, we can handle the PDF (and EPUB, I don't know what Amazon needs) via

    ```{=html}
    <iframe ...> ...
    ```
    ```{=pdf}
    <https://www.youtube.com/...>
    ```

And, this could also be a function inside `JDS` of course which handles the output if you like.

Reply:
I really love embedded content also!
I will add this to the issue (and project kanban dashboard) of the Amazon.com print and digital release.